File: webauthn-server-core/src/main/scala/com/yubico/webauthn/impl/NoneAttestationStatementVerifier.scala
package com.yubico.webauthn.impl

import com.yubico.webauthn.AttestationStatementVerifier
import com.yubico.webauthn.data.AttestationObject
import com.yubico.webauthn.data.ArrayBuffer
import com.yubico.webauthn.data.AttestationType
import com.yubico.webauthn.data.NoneAttestation


object NoneAttestationStatementVerifier extends AttestationStatementVerifier {

  override def getAttestationType(attestation: AttestationObject): AttestationType = NoneAttestation

  override def verifyAttestationSignature(attestationObject: AttestationObject, clientDataJsonHash: ArrayBuffer): Boolean = true

}

File: webauthn-server-core/src/main/java/com/yubico/webauthn/impl/NoneAttestationStatementVerifier.java
package com.yubico.webauthn.impl;

import com.yubico.webauthn.AttestationStatementVerifier;
import com.yubico.webauthn.data.AttestationObject;
import com.yubico.webauthn.data.AttestationType;


public class NoneAttestationStatementVerifier implements AttestationStatementVerifier {

    @Override
    public AttestationType getAttestationType(AttestationObject attestation) {
        return AttestationType.NONE;
    }

    @Override
    public boolean verifyAttestationSignature(AttestationObject attestationObject, byte[] clientDataJsonHash) {
        return true;
    }

}
